core/trino-main/src/main/java/io/trino/server/protocol/spooling/CoordinatorSegmentResource.java

```diff
@@ -113,10 +113,14 @@ public static UriBuilder spooledSegmentUriBuilder(ExternalUriInfo info)
                 .path(CoordinatorSegmentResource.class);
     }
 
-    public HostAddress nextActiveNode()
+    public HostAddress randomActiveWorkerNode()
     {
-        List<InternalNode> internalNodes = ImmutableList.copyOf(nodeManager.getActiveNodesSnapshot().getAllNodes());
-        verify(!internalNodes.isEmpty(), "No active nodes available");
+        List<InternalNode> internalNodes = nodeManager.getActiveNodesSnapshot().getAllNodes()
+                .stream()
+                .filter(node -> !node.isCoordinator())
```

Comment:
what if scheduling work on coordinator is enabled?

Reply:
I don't care :)

Reply:
this would redirect to itself and this would loop so I don't want it anyway

Reply:
it would not - you have different endopint now :P